roles/userspace-cni/tasks/main.yml

```diff
@@ -0,0 +1,150 @@
+---
+
+- name: "Add hugepages to sysctl.conf"
+  lineinfile:
+    path: /etc/sysctl.conf
+    regexp: 'nr_hugepages'
+    line: vm.nr_hugepages = 1024
+  register: set_hugepage_sysctl
+
+# You can verify with `cat /proc/meminfo | grep Huge`
+
+- name: Reload sysctl
+  shell: >
+    sysctl -p
+  when: set_hugepage_sysctl.changed
+
+# VPP installation
+# sudo yum install centos-release-fdio
+# sudo yum install vpp*
+
+- name: Install jq binary
+  get_url:
+    url: https://github.com/stedolan/jq/releases/download/jq-1.5/jq-linux64
+    dest: /usr/bin/jq
+
+- name: set jq binary permission
+  file:
+    path: /usr/bin/jq
+    mode: 0755
+
+# Moving from CentOS NFV SIG repo to Nexus repo.
+
+# cat /etc/yum.repos.d/fdio-release.repo 
+# [fdio-release]
+# name=fd.io release branch latest merge
+# baseurl=https://nexus.fd.io/content/repositories/fd.io.centos7/
+# enabled=1
+# gpgcheck=0
+
+# sudo yum install vpp-18.07-release.x86_64 \
+# vpp-lib-18.07-release.x86_64 \
+# vpp-plugins-18.07-release.x86_64 \
+# vpp-devel-18.07-release.x86_64 \
+# vpp-api-python-18.07-release.x86_64 \
+# vpp-api-lua-18.07-release.x86_64 \
+# vpp-api-java-18.07-release.x86_64 \
+# vpp-selinux-policy-18.07-release.x86_64
+
+# Dockerhub has been update with (both tags point to same image):
+#    bmcfall/vpp-centos-userspace-cni:0.3.0
+#    bmcfall/vpp-centos-userspace-cni:latest
+
+
+# FYI: On my server, I am almost out of hugepages, so you are at your limit. If you need more VMs, may need to reduce the number of hugepages given to each VM (like from 4K to 2K).
+# $ cat /proc/meminfo | grep -i huge
+# AnonHugePages:     79872 kB
+# HugePages_Total:   32768
+# HugePages_Free:     3991
+# :
+
+# Billy
+
+- name: Add multiple repositories into the same file (1/2)
+  yum_repository:
+    name: fdio-release
+    description: "fd.io repo"
+    baseurl: https://nexus.fd.io/content/repositories/fd.io.centos7/
+    gpgcheck: no
+
+- name: Install VPP repo
+  yum:
+    name: "centos-release-fdio"
+    state: latest
+
+# How do I install the proper VPP version?
+- name: Install VPP
+  yum:
```

Comment:
As previously, this is cleaner with the `vars` approach to allow better dependency resolution with the added benefit that it'll be like 1000 times faster.